### PR TITLE
Add Cilium Features to docs

### PR DESCRIPTION
**Pull request by chrisnegus**
*Description of changes:* Added Cilium features (EKS Anywhere, Open Source, and Cilium Enterprise) to the docs. The content is taken from the [Cilium in EKS Anywhere](https://isovalent.com/blog/post/cilium-eks-anywhere/#h-feature-availability) blog. This PR replaces [PR #5059](https://github.com/aws/eks-anywhere/pull/5059).

### CODE REVIEW

File: docs/content/en/docs/clustermgmt/networking/networking-and-security.md [hunk 1]
```diff
@@ -25,6 +25,26 @@ This section provides information about:
 * Using Cilium to securing workload connectivity using Kubernetes Network Policy.
 
 
+## Cilium Features
+The following table lists Cilium features and notes which of those features are built into EKS Anywhere, upstream Open Source, and Cilium Enterprise.
+
+<details><summary>Expand to see Cilium Features</summary>
+
+| Headline/Feature   | &nbsp;&nbsp;AWS EKS Anywhere | &nbsp;&nbsp;Cilium OSS |  &nbsp;&nbsp;Isovalent Cilium Enterprise |
```
Comment (csplinter) [inline docs/content/en/docs/clustermgmt/networking/networking-and-security.md:33]: Bit of a nit but suggest using "EKS Anywhere" or "EKS Anywhere Default Cilium" as the col name

File: docs/content/en/docs/clustermgmt/networking/networking-and-security.md [hunk 2]
```diff
@@ -25,6 +25,26 @@ This section provides information about:
 * Using Cilium to securing workload connectivity using Kubernetes Network Policy.
 
 
+## Cilium Features
+The following table lists Cilium features and notes which of those features are built into EKS Anywhere, upstream Open Source, and Cilium Enterprise.
```
Comment (csplinter) [inline docs/content/en/docs/clustermgmt/networking/networking-and-security.md:29]: built into EKS Anywhere's default Cilium version

File: docs/content/en/docs/clustermgmt/networking/networking-and-security.md (hunk 1 above)
Comment (chrisnegus) [reply docs/content/en/docs/clustermgmt/networking/networking-and-security.md:33]: Done.

**Comment by chrisdoherty4** [comment]
@Cajga Thanks for the feedback. We're aligning with what Isovalent advertise so we'll engage them and work through the inconsistencies.

**Review by chrisdoherty4 — APPROVED** [review]

**Pull request merged**